jsettlers.graphics/src/jsettlers/graphics/map/controls/original/panel/content/DistributionPanel.java

```diff
@@ -0,0 +1,286 @@
+        private static final float textMarginBottom = rowTextMarginBottom_px / rowHeight_px;
+        private static final float textPercentageWidth = 30f / 84f;
+
+        Label lblPercentage;
+        BarFill barFill;
+
+        public ConfigurationPanelRow(String receiver)
+        {
+            Label rowTitle = new Label(Labels.getString(receiver), EFontSize.SMALL);
+            rowTitle.setHorizontalAlignment(HorizontalAlignment.LEFT);
+
+            lblPercentage = new Label("0%", EFontSize.NORMAL);
+            lblPercentage.setHorizontalAlignment(HorizontalAlignment.LEFT);
+
+            barFill = new BarFill();
+
+            addChild(rowTitle, 0f, 1f - textHeight, 1f, 1f);
+            addChild(lblPercentage, 0f, 0f, textPercentageWidth, 1f - (textHeight + textMarginBottom));
+            addChild(barFill, textPercentageWidth, 0f, 1f, 1f - (textHeight + textMarginBottom));
+        }
+
+        public void setPercentage(int percentage)
+        {
+            lblPercentage.setText(Integer.toString(percentage)+"%");
+        }
+    }
+
+    private static class ConfigurationPanel extends UIPanel
+    {
+        private static final float panelHeight_px = contentHeight_px - marginTop_px;
+        private static final float rowHeight = rowHeight_px / panelHeight_px;
+        private static final float rowSpacing = rowSpacingV_px / panelHeight_px;
+
+        public ConfigurationPanel(final ConfigurationPanelRow[] rows)
+        {
+            int percentage = 100 / rows.length;
+            float top = 1f;
+            for (ConfigurationPanelRow row : rows) {
+                row.barFill.setAction(new ExecutableAction() {
+                    @Override
+                    public void execute() {
+                        float total = 0f;
+                        for(ConfigurationPanelRow r : rows){
+                            total += r.barFill.getValue();
+                        }
+                        for(ConfigurationPanelRow r : rows){
+                            int percentage = (int)(100 * (r.barFill.getValue() / total));
+                            r.setPercentage(percentage);
+                        }
+                    }
+                });
+                row.setPercentage(percentage);
+                addChild(row, 0f, top - rowHeight, 1f, top);
+                top -= rowHeight + rowSpacing;
+            }
+        }
+    }
+
+    private static class Icon extends UIPanel
+    {
+        ExecutableAction actionListener;
+
+        private Icon(ImageLink imageLink)
+        {
+            setBackground(imageLink);
+
+        }
+
+        public void setActionListener(ExecutableAction actionListener)
+        {
+            this.actionListener = actionListener;
+        }
+
+        @Override
+        public Action getAction(float relativex, float relativey)
+        {
+            return actionListener;
+        }
+    }
+
+    private enum Item {
```

Comment:
I know this is mostly for testing. We definitely need a way to get this from the game logic.